### ISSUE TITLE
Public members defined in a private constructor are ignored when --excludePrivate is set

### ISSUE THREAD

**Issue by stephanemagnenat** [TypeStrong/typedoc#1036]
When a constructor is private (to be used through a factory method), but its members are public, these do not get documented, although they are part of the public interface of the object. See this example:
```
class A {
    static create(b: number) {
        return new A(b);
    }
    /*
     * constructor
     * @param b public member b of class A
     */
    private constructor(public b: number) {
    }
}
```
In that case, member `b` in class `A` does not get documented when typedoc is run with `--excludePrivate`.

- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Expected Behavior
The member `b` should be documented.

## Actual Behavior
The member `b` is not documented, supposedly because typedoc sees that the constructor is private and stops looking into it.

## Steps to reproduce the bug
It should be pretty easy to reproduce. In my case the class is in a module with export, and I am using the following command line for Typedoc:
`typedoc --ignoreCompilerErrors --excludeNotExported --excludePrivate --out doc`

## Environment
  - Typedoc version: TypeDoc 0.14.2, using TypeScript 3.2.4
  - Node.js version: v10.10.0
  - OS: Ubuntu 16.04.6 LTS

**Comment by alalonde** [comment]
Ditto with protected constructors and `--excludeProtected`

**Comment by Gerrit0** [comment]
Fixed with 5d41a2e03a4cefa29ae84e7f9ef409d976d4ac56
